src/numerics/CVodesIntegrator.cpp

```diff
@@ -592,6 +645,59 @@ int CVodesIntegrator::nEvals() const
     return ne;
 }
 
+AnyMap CVodesIntegrator::nonlinearSolverStats() const
+{
+    // long int stats array to interface with cvodes
+    long int stats_arr[8];
+    long int* stats_ptr;
+    stats_ptr = stats_arr;
+    // AnyMap to return stats
+    AnyMap stats;
+    #if CT_SUNDIALS_VERSION >= 40
+        CVodeGetNonlinSolvStats(m_cvode_mem, stats_ptr, stats_ptr+1);
+    #else
+        throw CanteraError("CVodesIntegrator::nonLinearSolverStats", "Function not supported with sundials versions less than 4.");
+    #endif
+    // Add values to AnyMap
+    stats["nonlinear_iters"] = stats_arr[0];
+    stats["nonlinear_conv_fails"] = stats_arr[1];
```

Comment:
A more self-documenting approach that doesn't involve fighting so much with pointers and offsets:
```suggestion
    long int iters, conv_fails;
    // AnyMap to return stats
    AnyMap stats;
    #if CT_SUNDIALS_VERSION >= 40
        CVodeGetNonlinSolvStats(m_cvode_mem, &iters, &conv_fails);
    #else
        throw CanteraError("CVodesIntegrator::nonLinearSolverStats", "Function not supported with sundials versions less than 4.");
    #endif
    // Add values to AnyMap
    stats["nonlinear_iters"] = iters;
    stats["nonlinear_conv_fails"] = conv_fails;
```
This should also be used below for linearSolverStats